modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/model/listener/UserGroupModelListener.java

```diff
@@ -75,13 +79,42 @@ private void _unsubscribeUserGroupUsersFromGroup(
 			long groupId, long userGroupId)
 		throws PortalException {
 
-		for (long userId :
-				_userGroupLocalService.getUserPrimaryKeys(userGroupId)) {
+		Set<Long> userGroupUserIds = SetUtil.fromArray(
+			_userGroupLocalService.getUserPrimaryKeys(userGroupId));
 
-			if (!_groupLocalService.hasUserGroup(userId, groupId)) {
-				_blogsEntryLocalService.unsubscribe(userId, groupId);
+		userGroupUserIds.removeAll(
+			SetUtil.fromArray(_groupLocalService.getUserPrimaryKeys(groupId)));
+
+		long[] organizationIds = _groupLocalService.getOrganizationPrimaryKeys(
+			groupId);
+
+		Group group = _groupLocalService.getGroup(groupId);
+
+		if (group.getOrganizationId() > 0) {
+			organizationIds = ArrayUtil.append(
+				organizationIds, group.getOrganizationId());
+		}
```

Comment:
Instead of creating a fresh array and discard the old one when orgId > 0, it would be better to remove the group ID directly and avoid creating a whole new array just to make space for one additional element:
```
for (long organizationId : organizationIds) {
    userGroupUserIds.removeAll(...);
}

if (group.getOrganizationId() > 0) {
    userGroupUserIds.remove(group.getOrganizationId());
}
```